File: alfa/src/main/java/com/proyecto_analisis/alfa/model/entity/Usuario.java
package com.proyecto_analisis.alfa.model.entity;

import java.sql.Date;
import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Table(name = "usuario")
@Entity
@Data
@JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)

    @Column(name = "IdUsuario")
    private String idUsuario;

    @Column(name = "Nombre")
    private String nombre;

    @Column(name = "Apellido")
    private String apellido;

    @Column(name = "FechaNacimiento")
    private Date fechaNacimiento;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "IdStatusUsuario")
    private StatusUsuario idStatusUsuario;

    @Column(name = "Password")
    private String password;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "IdGenero")
    private Genero idGenero;

    @Column(name = "UltimaFechaIngreso")
    private LocalDateTime ultimaFechaIngreso;

    @Column(name = "IntentosDeAcceso")
    private Integer intentosDeAcceso;

    @Column(name = "SesionActual")
    private String sesionActual;

    @Column(name = "UltimaFechaCambioPassword")
    private Date ultimaFechaCambioPassword;

    @Column(name = "CorreoElectronico")
    private String correoElectronico;

    @Column(name = "RequiereCambiarPassword")
    private Integer requiereCambiarPassword;

    @Column(name = "Fotografia")
    private String fotografia;

    @Column(name = "TelefonoMovil")
    private String telefonoMovil;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "IdSucursal")
    private Sucursal idSucursal;

    @Column(name = "Pregunta")
    private String pregunta;

    @Column(name = "Respuesta")
    private String respuesta;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "IdRole")
    private Role idRole;

    @Column(name = "FechaCreacion")
    private LocalDateTime fechaCreacion;

    @Column(name = "UsuarioCreacion")
    private String usuarioCreacion;

    @Column(name = "FechaModificacion")
    private Date fechaModificacion;

    @Column(name = "UsuarioModificacion")
    private String usuarioModificacion;

}

File: angularprj/frontalm5/src/app/entity/usuario.ts
import { Genero } from './genero';
import { Role } from './Role';
import { StatusUsuario } from './statusUsuario';
import { Sucursal } from './Sucursal';

export class Usuario {

  idUsuario!: string;
  nombre!: string;
  apellido!: string;
  fechaNacimiento!: Date;
  idStatusUsuario!: StatusUsuario;
  password!: string;
  idGenero!: Genero;
  ultimaFechaIngreso!: Date;
  intentosDeAcceso!: Number;
  sesionActual!: string;
  ultimaFechaCambioPassword!: Date;
  correoElectronico!: string;
  requiereCambiarPassword!: Number;
  fotografia!: String;
  telefonoMovil!: String;
  idSucursal!: Sucursal;
  pregunta!: String;
  respuesta!: String;
  idRole!: Role;
  fechaCreacion!: Date;
  usuarioCreacion!: String;
  fechaModificacion!: Date;
  usuarioModificacion!: String;
}
